packages/kbn-search-response-warnings/src/components/search_response_warnings/i18n_utils.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import type { SearchResponseWarning } from '../../types';
+
+export const viewDetailsLabel = i18n.translate('searchResponseWarnings.viewDetailsButtonLabel', {
+  defaultMessage: 'View details',
+  description: 'View warning details button label',
+});
+
+export function getNonSuccessfulClusters(warnings: SearchResponseWarning[]) {
+  const nonSuccessfulClusters = new Set<string>();
+  warnings.forEach((warning) => {
+    Object.keys(warning.clusters).forEach((clusterName) => {
+      if (warning.clusters[clusterName].status !== 'successful') {
+        nonSuccessfulClusters.add(clusterName);
+      }
+    });
+  });
+  return nonSuccessfulClusters;
+}
+
+export function getWarningsTitle(warnings: SearchResponseWarning[]) {
+  const nonSuccessfulClusters = getNonSuccessfulClusters(warnings);
+  const clustersClause = i18n.translate('searchResponseWarnings.title.clustersClause', {
+    defaultMessage:
+      'Problem with {nonSuccessfulClustersCount} {nonSuccessfulClustersCount, plural, one {cluster} other {clusters}}',
+    values: { nonSuccessfulClustersCount: nonSuccessfulClusters.size },
+  });
+
+  return warnings.length <= 1
+    ? clustersClause
+    : i18n.translate('searchResponseWarnings.title.clustersClauseAndRequestsClause', {
+        defaultMessage: '{clustersClause} in {requestsCount} requests',
+        values: {
+          clustersClause,
+          requestsCount: warnings.length,
+        },
+      });
+}
+
+export function getWarningsDescription(
+  warnings: SearchResponseWarning[],
+  visualizationLabel?: string
+) {
+  const nonSuccessfulClusters = getNonSuccessfulClusters(warnings);
+  return i18n.translate('searchResponseWarnings.description', {
+    defaultMessage:
+      '{nonSuccessfulClustersCount, plural, one {This cluster} other {These clusters}} had issues returning data. This might result in an incomplete {visualizationLabel}.',
+    values: {
+      nonSuccessfulClustersCount: nonSuccessfulClusters.size,
+      visualizationLabel: visualizationLabel
```

Comment:
Concatenating phrases might not work for some languages. Would be better to allow customizing the whole last sentence. Or we could use a more generic `This might result in incomplete data.` instead.

Reply:
"This might result in incomplete data" won't work because it sounds to similiar to the next sentence "This might result in an incomplete".  

What about creating2 separate i18n strings for the 2 cases so other languages and re-write the entire structure to what fits best?
1) searchResponseWarnings.description.singleCluster: "This cluster had issues returning data. This might result in an incomplete {visualizationLabel}."
2) searchResponseWarnings.descriptionmultipleClusters: "These clusters had issues returning data. This might result in an incomplete {visualizationLabel}."

Reply:
I like the idea of separating messages for the info about clusters because in some languages a verb should be different based on 1 vs multiple items.

I was talking about the issue with using a variable for the last noun in `This might result in an incomplete {visualizationLabel}.` 
For example, in German different nouns require different articles and adjectives. It's not possible to translate the beginning of a sentence if the remainder is unknown to the translator and vice versa.

Reply:
@gchaps thoughts on changing "This might result in an incomplete {visualizationLabel}." to "This might result in incomplete data."?

Reply:
How about:

> This cluster had issues returning data and results might be incomplete.